sdk/resource/builtin.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.opentelemetry.io/otel/label"
+	opentelemetry "go.opentelemetry.io/otel/sdk"
+	"go.opentelemetry.io/otel/semconv"
+)
+
+type (
+	TelemetrySDK struct{}
+	Host         struct{}
```

Comment:
Can we add some docs to the exported types and functions?

```suggestion
	// TelemetrySDK is a Detector that provides information about the OpenTelemetry SDK used.
	// This Detector is included as a builtin. If this information is not wanted it needs to explicitly be
	// configured to not be included.
	TelemetrySDK struct{}
	// Host is a Detector that provides information about the host being run on. This Detector is
	// included as a builtin. If this information is not wanted it needs to explicitly be configured to
	// not be included.
	Host         struct{}
```